### PR TITLE
Update chloropleths with new datasets

### PR DESCRIPTION
**Pull request by lizgzil**
---

# Description

`green_measures_itl_2_code_chloropleth.html` couldn't be updated using the latest dataset - not sure why but there was a bug

Fixes # (issue)

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 1]
```diff
@@ -564,6 +564,10 @@ def create_agg_data(
     """
     Much like create_agg_occ_measures but more generic to aggregate by any column
     """
```
Comment (lizgzil) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:566]: @india-kerle please could you check the changes to this script are what we planned to do

Comment (india-kerle) [reply dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:566]: taking a look !

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 2]
```diff
@@ -630,12 +634,10 @@ def create_agg_data(
                 )
 
             top_5_sics_num = (
-                filtered_data[["SIC_2_digit", "SIC_2_digit_name"]]
-                .value_counts()[0:5]
-                .to_dict()
+                filtered_data[["SIC", "SIC_name"]].value_counts()[0:5].to_dict()
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:637]: thanks for this liz

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 3]
```diff
@@ -701,15 +714,13 @@ def create_agg_data(
                 "top_5_green_skills": green_skill_info,
                 "top_5_not_green_skills": not_green_skill_info,
                 # Industry
-                "num_unique_SIC2": filtered_data["SIC_2_digit"].nunique(),
-                "num_null_sic2": len(
-                    filtered_data[pd.isnull(filtered_data["SIC_2_digit"])]
-                ),
-                "num_top_sic2": filtered_data["SIC_2_digit"].value_counts()[0]
-                if len(filtered_data["SIC_2_digit"].value_counts()) > 0
+                "num_unique_SIC": filtered_data["SIC"].nunique(),
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:717]: i guess we should use SIC_name given the nans?

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 4]
```diff
@@ -701,15 +714,13 @@ def create_agg_data(
...
+                "num_null_sic": len(filtered_data[pd.isnull(filtered_data["SIC"])]),
+                "num_top_sic": filtered_data["SIC"].value_counts()[0]
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:719]: ditto lets use SIC-name

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 5]
```diff
@@ -701,15 +714,13 @@ def create_agg_data(
...
+                if len(filtered_data["SIC"].value_counts()) > 0
                 else None,
-                "num_other_sic2": sum(filtered_data["SIC_2_digit"].value_counts()[1:])
-                if len(filtered_data["SIC_2_digit"].value_counts()) > 1
+                "num_other_sic": sum(filtered_data["SIC"].value_counts()[1:])
+                if len(filtered_data["SIC"].value_counts()) > 1
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:723]: ditto SIC_name

File: dap_prinz_green_jobs/config/ojo_analysis.yaml [hunk 6]
```diff
@@ -4,5 +4,5 @@ config: "base"
 skills_date_stamp: "20231204"
 occ_date_stamp: "20231127"
 ind_date_stamp: "20231205"
-data_type: "mixed"
```
Comment (india-kerle) [inline dap_prinz_green_jobs/config/ojo_analysis.yaml:7]: good spot 🤦

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py [hunk 7]
```diff
@@ -564,6 +564,10 @@ def create_agg_data(
     """
     Much like create_agg_occ_measures but more generic to aggregate by any column
     """
+    prop_job_ads_per_itl2_all = (
+        all_green_measures_df["itl_2_name"].value_counts(normalize=True).to_dict()
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:568]: does normalizing do anything here that we don't want? my hunch says no but just triple checking

File: dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py [hunk 8]
```diff
@@ -20,7 +20,6 @@
 import os
 
 if __name__ == "__main__":
-
```
Comment (india-kerle) [inline dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py:23]: just ran this script with no issues - i will say it took ~17 minutes to run so maybe we can make note in the README that it takes a while!

**Review by india-kerle — CHANGES_REQUESTED** [review]
Thanks for this liz!

I reran your scripts and encountered no bugs plus reviewed the process_ojo_measures.py methods - my one thing is given the SIC nans i think we should rely on SIC names for everything (until this is investigated further). Another thing is i took a look at the github actions and it looks like numpy has since fixed their bug and we're now faced with a permissions issue - let's not address in this PR but a low priority nice thing to fix - i'll make an issue for it.

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py (hunk 3 above)
Comment (lizgzil) [reply dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:717]: it was actually the name which had the nan's in!

File: dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py (hunk 7 above)
Comment (lizgzil) [reply dap_prinz_green_jobs/analysis/ojo_analysis/process_ojo_green_measures.py:568]: no - it just gives you the proportions not the counts

**Comment by india-kerle** [comment]
@lizgzil I made a little change to your regional analysis notebooks and was able to product the `green_measures_itl_2_code_chloropleth.html` graph - do take a quick look about my changes but I think we're in business!

**Comment by lizgzil** [comment]
thanks @india-kerle - this looks good - I've updated the images in the viz folder

**Pull request merged**